fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/generator/metric/ProtocolLogicalMetricGenerator.java

```diff
@@ -159,18 +160,31 @@ public void validate(
         List<String> invalidMetricNames = new ArrayList<>();
 
         for (ApiMetric metric : apiMetrics) {
+
+            // if base metric isn't available just log and move on
+            LogicalMetric baseLogicalMetric = metricDictionary.get(metric.getBaseApiMetricId());
+            if (baseLogicalMetric == null) {
+                invalidMetricNames.add(metric.getRawName());
+                continue;
+            }
+
+            // if no name change nor parameters to apply just add to resolved metrics and continue
+            if (metric.getRawName().equals(metric.getBaseApiMetricId()) &&
+                            metric.getParameters().isEmpty()) {
+                metrics.add(baseLogicalMetric);
```

Comment:
This parameters is empty check will get blown up by the other pending PR where we annotate timegrain onto the metric. We need to remove it.

Reply:
I actually don't think this check is even necessary. If no protocols are applied to the metric and there is no name change, then the rename operation will be a noop.